public_html/wp-content/mu-plugins/blocks/assets/src/shared/featured-image/index.js

```diff
@@ -9,150 +9,56 @@ import classnames from 'classnames';
 const { Component } = wp.element;
 
 /**
- * Displays featured image, can be linked with block control for size. Fetches the smallest possible image size required to render the current size.
+ * Displays featured image, can be linked with block control for size.
  */
 export default class FeaturedImage extends Component {
 
 	/**
-	 * Constructor for component. Size can be provided, if height and width both are provided, then they take precedence over size.
-	 *
 	 * @param props Props for function.
-	 * @param props.wpMediaDetails Available sizes of images in the format as returned by WP API. This is the `sizes` object inside `media_details` inside `wp:featuredMedia` object. Not required if `props.availableSizes is present.
-	 * @param props.availableSizes Available sizes of image along with source URL. Not required if `props.wpMediaDetails` is present. Should be in this format:
-	 * [
-	 * 		{
-	 * 		 	height: 999,
-	 * 		 	width: 999,
-	 * 		 	source_url: https://link.to/image-size-1.png
-	 * 		},
-	 * 		{
-	 * 			height: 500,
-	 * 			width: 500,
-	 * 			source_url: https://link.to/image-size-2.png
-	 * 		}
-	 * 		...
-	 * ]
+	 * @param props.wpMediaDetails Available sizes of images in the format as returned by WP API. This is the `sizes` object inside `media_details` inside `wp:featuredMedia` object.
 	 * @param props.height Height in pixels for image.
 	 * @param props.width Width in pixels for image.
 	 * @param props.className Classname for image element
 	 * @param props.alt Alt text for image
-	 * @param props.onChange Function callback for when Image URL is changed. Current URL will be passed as a parameter.
 	 */
 	constructor( props ) {
-		super();
+		super( props );
```

Comment:
I intentionally removed `props` in 2e720880c751a70df2d48b6293e46dad77574e24. Is there a reason to add it back?

Reply:
I don't think it works correctly if you don't include `props` as a parameter in `super()`...

Reply:
It seems to work fine for me, and everything I read suggests that it's not necessary. Do you see anywhere that it doesn't work?

Reply:
My read is that if the parent constructor needs any of the parameters being fed to the child, then `super` should include it.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/super

There are probably some situations where that isn't the case, so it doesn't affect anything to omit it. But I would argue that for consistency's sake we should just always include it, unless there's a situation where we specifically _don't_ want the parent constructor to get the same parameters, in which case we would want to explain that in a code comment.

Reply:
Resolving this comment. Please re-open if any further discussion is needed.

Reply:
Marking as unresolved because right now it's inconsistent, most of the constructors don't have it, but this one does, so we should decide one way or the other and make them all consistent.

It seems like, so far, it's not needed in 100% of the situations, so it feels more logical to leave it off by default, and then pass it in when necessary.

It seems like Andrew [felt like it was unnecessary](https://github.com/WordPress/wordcamp.org/pull/13#discussion_r249476093), so this might make us more consistent w/ how Gutenberg does it. But actually, it looks like [G typically calls it with `...arguments`](https://github.com/WordPress/gutenberg/search?l=JavaScript&q=super%28), so maybe we should do that always instead of `props`?

Reply:
Andrew's comment was actually about binding the render function, further down, not about the `super` call.

But, I found this: https://stackoverflow.com/questions/30571875/whats-the-difference-between-super-and-superprops-in-react-when-using-e

...which would indicate that including `props` in the `super` call is only necessary if you need to refer to `this.props` elsewhere within the constructor method. **But**, it also points out that in the React docs, it says [this](https://reactjs.org/docs/state-and-lifecycle.html#adding-local-state-to-a-class):

> Class components should always call the base constructor with `props`.

Reply:
haha, yeah, that's the thing I read that made me feel like we should remove it :)

So it seems like it's pretty intangible. For the sake of not bikeshedding it, I'll just revert 2e72088 and we can move on to more important things :)

Reply:
Done in 494f7f32c4b2ec095b26cd184f5479c1a3ad1249